lib/easypost/services/smartrate.rb

```diff
@@ -0,0 +1,21 @@
+class EasyPost::Services::Smartrate < EasyPost::Services::Service
+  MODEL_CLASS = EasyPost::Models::Smartrate
+
+  # Retrieve the estimated delivery date of each carrier-service level combination via the
+  # Smart Deliver By API, based on a specific ship date and origin-destination postal code pair.
+  def estimate_delivery_date(params = {})
+    url = '/smartrate/deliver_by'
+
+    response = @client.make_request(:post, url, params)
+    EasyPost::InternalUtilities::Json.convert_json_to_object(response, MODEL_CLASS)
+  end
+
+  # Retrieve a recommended ship date for each carrier-service level combination via the
+  # Smart Deliver On API, based on a specific delivery date and origin-destination postal code pair.
+  def recommend_ship_date(params = {})
+    url = '/smartrate/deliver_on'
```

Comment:
```suggestion
    url = 'smartrate/deliver_on'
```

Not sure if it matters, but just in case and for consistency.

Reply:
I have removed the extra slash to match the consistency